scrycall/scry_args.py

```diff
@@ -0,0 +1,68 @@
+import argparse
+from .scry_output import Attributes
+
+
+class ArgumentError(Exception):
+    """Error with input arguments"""
+
+
+def parse_args(args: list) -> argparse.Namespace:
+    """Parse command line arguments
+
+    Args:
+        args: sys argv
+
+    Returns:
+        parsed_args: Args namespace
+    """
+    parser = _parser()
+    parsed_args = parser.parse_args(args)
+
+    parsed_args.query = " ".join(parsed_args.query)
+    parsed_args.query.replace("`", "'")
+
+    if not parsed_args.query:
+        parser.print_help()
+        raise ArgumentError("No query provided!")
+
+    return parsed_args
+
+
+def _parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(
+        description="Scryfall CLI",
+        epilog="""
+        %n    name
+        %m    mana_cost
+        %c    cmc  (converted mana cost)
+        %y    type_line
+        %p    power
+        %t    toughness
+        %l    loyalty
+        %o    oracle_text
+        %f    flavor_text
+        %%    this will print a literal % instead of interpreting a special character
+        %|    this will separate output into nicely spaced columns
+        """,
+    )
+    parser.add_argument(
+        "query",
+        nargs="*",
+        metavar="query",
+        default=[],
+        help="Scryfall query",
+    )
+    parser.add_argument(
+        "-f",
```

Comment:
I want to avoid arguments with only a single dash. Scryfall syntax uses `-` as a negation. So typing `scry -f` would query for all cards without an "f" in their name. To keep things as unambiguous as possible, I wanted to only use `--` to distinguish optional parameters.

This however leads to another problem, as `--` is also a legal search term meaning "cards without a dash in their name." More thought needs to go into this, but for now I'd like to stick to double dashes.

Reply:
hmm, that is a tricky problem; compounded that double dashes don't help either as `--format "%n` is a legal command line format. In reality, you CAN search for literally anything in an open-text query box so IMO letting `--` be the delimiter is acceptable.

I'd have to think a little more about what could work well.